docs/remote-clusters.asciidoc

```diff
@@ -0,0 +1,125 @@
+
+. Ensure that both clusters trust each others certificate authority.
+. Configure the remote cluster connection via the Elasticsearch REST API.
+
+For the purposes of this guide we use the following running example:
+
+* `cluster_one` resides inside Kubernetes and is managed by ECK
+* `cluster_two` is not hosted inside the same Kubernetes cluster as `cluster_one` and may even not be managed by an ECK instance
+
+We will configure `cluster_one` as a remote cluster in `cluster_two`.
+
+
+===== Ensure both clusters trust each others certificate authority
+
+The certificate authority used by ECK to issue certificates for the Elasticsearch transport layer is stored in a secret named `<cluster_name>-es-transport-certs-public`. Extract the certificate for `cluster_one` as follows:
+
+[source,sh]
+----
+kubectl get secret cluster_one-es-transport-certs-public \
+-o go-template='{{index .data "ca.crt" | base64decode}}' > remote.ca.crt
+----
+
+You then need to configure the CA as one of the trusted CAs in `cluster_two`. If that cluster is hosted outside of Kubernetes simply add the CA you extracted to the list of CAs in link:https://www.elastic.co/guide/en/elasticsearch/reference/current/security-settings.html#_pem_encoded_files_3[`xpack.security.transport.ssl.certificate_authorities`]
+
+If `cluster_two` is also managed by an ECK instance, proceed as follows:
+
+Create a secret with the CA certificate you just extracted:
+[source,sh]
+----
+kubectl create secret generic remote-certs --from-file=remote.ca.crt
+----
+
+Use this secret to configure `cluster_one`'s CA as a trusted CA in `cluster_two`:
+
+[source,yaml,subs="attributes"]
+----
+apiVersion: elasticsearch.k8s.elastic.co/{eck}
+kind: Elasticsearch
+metadata:
+  name: cluster_two
+spec:
+  nodeSets:
+  - config:
+      xpack.security.transport.ssl.certificate_authorities:
+      - /usr/share/elasticsearch/config/other/remote.ca.crt
+    count: 3
+    name: default
+    podTemplate:
+      spec:
+        containers:
+        - name: elasticsearch
+          volumeMounts:
+          - mountPath: /usr/share/elasticsearch/config/other
+            name: remote-certs
+        volumes:
+        - name: remote-certs
+          secret:
+            secretName: remote-certs
+  version: 7.6.0
+----
+
+Repeat the steps with reversed roles adding the CA of `cluster_two` to `cluster_one`.
+
+===== Configure the remote cluster connection via the Elasticsearch REST API
+
+Expose the transport layer of `cluster_one` to `cluster_two` which is residing outside of `cluster_one`'s Kubernetes cluster.
+
+[source,yaml]
+----
+apiVersion: v1
+kind: Service
+metadata:
+  name: cluster_one-es-transport
+spec:
+  selector:
+    common.k8s.elastic.co/type: elasticsearch
+    elasticsearch.k8s.elastic.co/cluster-name: cluster_one
+  type: LoadBalancer
```

Comment:
This will mostly work with cloud providers. I would add a note that it could also be done with an `Ingress` resource.